src/symbol/collision_feature.js

```diff
@@ -72,42 +72,60 @@ class CollisionFeature {
     _addLineCollisionBoxes(collisionBoxArray, line, anchor, segment, labelLength, boxSize, featureIndex, sourceLayerIndex, bucketIndex) {
         const step = boxSize / 2;
         const nBoxes = Math.floor(labelLength / step);
+        // We calculate line collision boxes out to 150% of what would normally be our
+        // max size, to allow collision detection to work on labels that expand as
+        // they move into the distance
+        const nPitchPaddingBoxes = Math.floor(nBoxes / 4);
```

Comment:
Yeah, although it makes collision detection slightly more responsive when you pitch the map, calculating these ahead of time probably isn't too important. The actual boxes are created at tile parsing time, so they need to be built regardless of pitch. However, at placement time, we could avoid inserting them into the collision grid in `CollisionTile::insertCollisionFeature`